parlai/agents/bart/modules.py

```diff
@@ -56,7 +56,19 @@ def reorder_decoder_incremental_state(
         """
         Incremental state is weird to handle when we seed decoder with two inputs
         initially.
-
-        We leave as a future exercise.
         """
-        return None
+        # we only have this method called when it's actually being used
+        assert incremental_state is not None
+        assert len(incremental_state) > 0
+
+        for layer in incremental_state.keys():
```

Comment:
Nit: could probably just do `for incr_state_l in incremental_state.values():` here, right?